dot/sync/syncer.go

```diff
@@ -346,6 +350,19 @@ func (s *Service) handleBlock(block *types.Block) error {
 		return fmt.Errorf("failed to execute block %d: %w", block.Header.Number, err)
 	}
 
+	insKeys, err := ts.GetInsertedNodeHashes()
+	if err != nil {
+		return fmt.Errorf("failed to get state trie inserted keys: block %s %w", block.Header.Number, err)
+	}
+
+	delKeys := ts.GetDeletedNodeHashes()
+
+	blockHash := block.Header.Hash()
+	err = s.pruner.StoreJournalRecord(delKeys, insKeys, &blockHash, block.Header.Number)
+	if err != nil {
+		return err
+	}
```

Comment:
instead of having the same code here and in babe, can you put this code into `storageState.StoreTrie` instead? I think it makes more sense to have it there, since that's where the nodes are being written to the db

Reply:
and then you wouldn't need to pass the pruner to sync and babe

Reply:
Done